docs/guides/cli.md

```diff
@@ -0,0 +1,293 @@
+  --version            show program's version number and exit
+  --log-level <level>  Log level to use for the nnbench package, defaults to NOTSET (no logging).
+
+Available commands:
+  
+    run                Run a benchmark workload.
+    compare            Compare results from multiple benchmark runs.
+```
+
+Supported log levels are `"DEBUG", "INFO", "WARNING", "ERROR"`, and `"CRITICAL"`.
+
+## Running benchmark workloads on the command line
+
+This is the responsibility of the `nnbench run` subcommand.
+
+```commandline
+$ nnbench run -h                                                                         
+usage: nnbench run [-h] [--context <key=value>] [-t <tag>] [-o <file>] [<benchmarks>]
+
+positional arguments:
+  <benchmarks>          Python file or directory of files containing benchmarks to run.
+
+options:
+  -h, --help            show this help message and exit
+  --context <key=value>
+                        Additional context values giving information about the benchmark run.
+  -t, --tag <tag>       Only run benchmarks marked with one or more given tag(s).
+  -o, --output-file <file>
+                        File or stream to write results to, defaults to stdout.
+```
+
+So to run a benchmark workload contained in a single `benchmarks.py` file, you would run `nnbench run benchmarks.py`.
+For tips on how to structure and annotate your benchmarks, refer to the [organization](organization.md) guide.
+
+For injecting context values on the command line, you need to give the key-value pair explicitly by passing the `--context` switch.
+For example, to look up and persist the `pyyaml` version in the current environment, you could run the following:
+
+```commandline
+nnbench run .sandbox/example.py --context=pyyaml=`python3 -c "from importlib.metadata import version; print(version('pyyaml'))"`
+```
+
+!!! tip
+    Both `--context` and `--tag` are appending options, so you can pass multiple context values and multiple tags per run.
+
+## Comparing results across multiple benchmark runs
+
+To create a comparison table between multiple benchmark runs, use the `nnbench compare` command.
+
+```commandline
+$ nnbench compare -h                                            
+usage: nnbench compare [-h] [-P <name>] [-C <name>] [-E <name>] records [records ...]
+
+positional arguments:
+  records               Records to compare results for. Can be given as local files or remote URIs.
+
+options:
+  -h, --help            show this help message and exit
+  -P, --include-parameter <name>
+                        Names of input parameters to display in the comparison table.
+  -C, --include-context <name>
+                        Context values to display in the comparison table. Use dotted syntax for nested context values.
+  -E, --extra-column <name>
+                        Additional record data to display in the comparison table.
+```
+
+Supposing we have the following records from previous runs:
```

Comment:
We could shorten this example to use only one variable, e.g. `add`. I feel using three makes this example pretty long without adding much additional information